File: src/main/java/dev/bernasss12/bebooks/util/text/IconTooltipDataText.java
package dev.bernasss12.bebooks.util.text;

import net.minecraft.item.ItemStack;
import net.minecraft.text.*;

import java.util.Collections;
import java.util.List;

public record IconTooltipDataText(List<ItemStack> icons) implements OrderedText, Text {

    @Override
    public boolean accept(CharacterVisitor visitor) {
        return false;
    }

    @Override
    public Style getStyle() {
        return Style.EMPTY;
    }

    @Override
    public String asString() {
        return "";
    }

    @Override
    public List<Text> getSiblings() {
        return Collections.emptyList();
    }

    @Override
    public MutableText copy() {
        return new LiteralText(asString());
    }

    @Override
    public MutableText shallowCopy() {
        return copy();
    }

    @Override
    public OrderedText asOrderedText() {
        return this;
    }
}

File: src/main/kotlin/dev/bernasss12/bebooks/util/text/IconTooltipDataText.kt
package dev.bernasss12.bebooks.util.text

import net.minecraft.item.ItemStack
import net.minecraft.text.*

data class IconTooltipDataText(val icons: List<ItemStack>) : OrderedText, Text {

    override fun accept(visitor: CharacterVisitor?): Boolean = false

    override fun getStyle(): Style = Style.EMPTY

    override fun getContent(): TextContent = Text.literal("").content

    override fun getSiblings(): MutableList<Text> = mutableListOf()

    override fun asOrderedText(): OrderedText = this
}